pkg/id/id.go

```diff
@@ -94,19 +94,21 @@ func NewAllocator(params *AllocatorParams) Allocator {
 }
 
 // Alloc returns a new id.
-func (alloc *allocatorImpl) Alloc() (uint64, error) {
+func (alloc *allocatorImpl) Alloc(count uint32) (uint64, uint32, error) {
 	alloc.mu.Lock()
 	defer alloc.mu.Unlock()
 
-	if alloc.base == alloc.end {
-		if err := alloc.rebaseLocked(true); err != nil {
-			return 0, err
+	for range count {
+		if alloc.base == alloc.end {
+			if err := alloc.rebaseLocked(true); err != nil {
+				return 0, 0, err
+			}
 		}
-	}
 
-	alloc.base++
+		alloc.base++
+	}
 
-	return alloc.base, nil
+	return alloc.base, count, nil
```

Comment:
the result of count can be skipped

Reply:
The Alloc needs the `count` as the return value in the current implementation. We do have another way that does not add count in the response but I prefer to leave it.